scraping_flows.py

```diff
@@ -0,0 +1,48 @@
+import json
+import pandas as pd
+from scraping_utils import *
+
+
+#############################################################################
+# FLOWS
+#############################################################################
+
+
+#@TODO move to scraping_flows.py
+def get_course_tables(links):
+    '''
+    Returns a dictionary of pandas DataFrames of courses from each of the links provided in links
+    The keys are the subject code portion of the link (the last bit separated by /)
+    Used in get_subjects.ipynb
+    '''
+    course_tables = dict()
+    for link in links:
+        code = link.strip('/').rsplit('/')[-1]
+        try:
+            course_tables[code] = get_courses(link)
+        except AttributeError as e:
+            print("Attribute Error:")
+            print(link)
+            print(e)
+            print("Continuing...")
+    return course_tables
+
+#@TODO move these two reading functions elsewhere?
```

Comment:
I'm not sure what the goal of this `scraping_flows.py` file is I see that it houses some of the old functions that don't really fit in elsewhere in the new structure, but it's also not imported by any other files so I'm a bit confused as to what it's doing. Maybe it's because I'm not sure what you mean by a "flow"??

Reply:
Ahh, so you're sort of on the right track.  Basically the goal was to separate modular tools which I wrote for the scraping, and higher level 'work flows' which use said tools to actually do the scraping on a pre-defined (as opposed to user-defined) scale.  That being said, this isn't really essential or core to the functionality, just something I wanted to keep track of.  It's also not very mature yet, so maybe stay tuned for more?

Reply:
Ok so if I understand correctly then essentially this file will house all the functions that the user would be interested in calling, and by extension is essentially what our api endpoints need to map too?? If yes then I think this is a great move!

Reply:
More or less, yes.  Also some "automation" for example to just go a scrape all the course info and save it somewhere.